[4.8.1] app.config.ts: New variants no longer merged into component props types

### Environment

- Operating System: Linux
- Node Version:     v20.12.0
- Nuxt Version:     4.2.1
- CLI Version:      3.30.0
- Nitro Version:    2.12.9
- Package Manager:  pnpm@8.15.6
- Builder:          -
- User Config:      modules, devtools, compatibilityDate, future, css
- Runtime Modules:  @nuxt/ui@4.8.1
- Build Modules:    -

### Is this bug related to Nuxt or Vue?

Nuxt

### Package

v4.x

### Version

v4.8.1

### Reproduction

https://codesandbox.io/p/devbox/elated-chihiro-lry3c6

Run `pnpm typecheck`.

### Description

Hello,
since v4.8.1 adding a new non-default variant defined via `app.config.ts` causes type errors on consumers. E.g. if I want to create a new larger size for UAvatar I could do it like this before:

```ts
// app.config.ts
export default defineAppConfig({
  ui: {
    avatar: {
      variants: {
          size: {
              '4xl': { root: 'size-20 text-2xl' },
          },
      },
    },
  },
});
```

Since v4.8.1 this new size is not a valid value anymore and only the nuxt-ui default values are accepted:
```
error TS2322: Type '"4xl"' is not assignable to type '"2xl" | "md" | "3xs" | "2xs" | "xs" | "sm" | "lg" | "xl" | "3xl" | undefined'.

3     <UAvatar size="4xl" />
```

The runtime behaviour is unchanged and correctly applies the defined values of the custom variant.

### Additional context

I believe #6520 introduced this regression, as commenting out the new `AppConfig` declaration in `.nuxt/types/ui.d.ts` fixes the problem.

### Logs

```shell-script

```

--- a/src/templates.ts
+++ b/src/templates.ts
@@ -323,7 +323,7 @@ type AppConfigUI = {
   tv?: typeof defaultConfig
 } & TVConfig<typeof ui>
 
-type AppConfigRuntimeUI = DeepRequired<Pick<AppConfigUI, 'colors' | 'icons' | 'tv'>> & Omit<AppConfigUI, 'colors' | 'icons' | 'tv'>
+type AppConfigRuntimeUI = DeepRequired<Pick<AppConfigUI, 'colors' | 'icons' | 'tv'>> & typeof ui
 
 declare module '@nuxt/schema' {
   interface AppConfigInput {
@@ -333,7 +333,7 @@ declare module '@nuxt/schema' {
      */
     ui?: AppConfigUI
   }
-  interface AppConfig {
+  interface CustomAppConfig {
     ui: AppConfigRuntimeUI
   }
 }

